src/neurotechdevkit/scenarios/_base.py

```diff
@@ -926,59 +918,97 @@ def render_layout(
+                points are in that material. Defaults to None.
+            origin (Optional[list[float]], optional): The location of the origin of the
+                scenario (in meters). Defaults to None.
+            sources (Optional[list[Source]], optional): The list of sources in the
+                scenario. Defaults to None.
+            material_outline_upsample_factor (Optional[int], optional): The factor by
+                which to upsample the material outline when rendering the scenario.
+                Defaults to None.
+            target (Optional[Target], optional): The target in the scenario. Defaults
+                to None.
+            problem (Optional[Problem], optional): The problem definition for the
+                scenario. Defaults to None.
+            grid (Optional[Grid], optional): The grid for the scenario. Defaults to
+                None.
+            slice_axis (Optional[SliceAxis], optional): The axis along which to slice
+                the 3D field to be recorded. If None, then the complete field will be
+                recorded. Use 0 for X axis, 1 for Y axis and 2 for Z axis. Defaults to
+                None.
+            slice_position (Optional[float], optional): The position (in meters) along
+                the slice axis at which the slice of the 3D field should be made.
+                Defaults to None.
+            viewer_config_3d (Optional[rendering.ViewerConfig3D], optional): The
+                configuration to use when rendering the 3D scenario. Defaults to None.
         """
-        pass
-
-    @property
-    @abc.abstractmethod
-    def viewer_config_3d(self) -> rendering.ViewerConfig3D:
-        """Configuration parameters for 3D visualization of this scenario."""
-        pass
+        if slice_axis is not None:
+            self.slice_axis = slice_axis
+        if slice_position is not None:
+            self.slice_position = slice_position
+        if viewer_config_3d is not None:
+            self.viewer_config_3d = viewer_config_3d
+        super().__init__(
+            center_frequency=center_frequency,
+            material_properties=material_properties,
+            material_masks=material_masks,
+            origin=origin,
+            sources=sources,
+            material_outline_upsample_factor=material_outline_upsample_factor,
+            target=target,
+            problem=problem,
+            grid=grid,
+        )
 
     def get_target_mask(self) -> npt.NDArray[np.bool_]:
         """Return the mask for the target region."""
         target_mask = create_grid_spherical_mask(
             grid=self.problem.grid,
-            origin=self.origin,
+            origin=np.array(self.origin, dtype=float),
             center=self.target_center,
             radius=self.target_radius,
         )
         return target_mask
 
     def simulate_pulse(
         self,
-        center_frequency: float = 5.0e5,
```

Comment:
Could also change this return type to `results.PulsedResult3D`

Reply:
nice catch!